lib/Semantics/check-declarations.cpp

```diff
@@ -707,6 +707,24 @@ void CheckHelper::CheckDerivedType(
         }
       }
     }
+    DerivedTypeSpec derived{symbol.name(), symbol};
+    derived.set_scope(*symbol.scope());
+    if (FindCoarrayUltimateComponent(derived) && // C736
+        !(parentDerived && FindCoarrayUltimateComponent(*parentDerived))) {
+      messages_.Say(
+          "Type '%s' has a coarray ultimate component so the type it extends "
+          "('%s') must have a coarray ultimate component"_err_en_US,
+          symbol.name(), parentDerived->typeSymbol().name());
+    }
+    if (FindEventOrLockPotentialComponent(derived) && // C737
+        !(FindEventOrLockPotentialComponent(*parentDerived) ||
+            IsEventTypeOrLockType(parentDerived))) {
+      messages_.Say(
+          "Since type '%s' has an EVENT_TYPE or LOCK_TYPE component, the type"
+          " it extends ('%s') must either be EVENT_TYPE or LOCK_TYPE or"
+          " contain an EVENT_TYPE or LOCK_TYPE component"_err_en_US,
+          symbol.name(), parentDerived->typeSymbol().name());
+    }
```

Comment:
These two errors are similar so they should use the same wording. Perhaps change the second one to:
"Type '%s' has an EVENT_TYPE or LOCK_TYPE component so the type it extends must...".
Also, "have" would be better than "contain" in the third line.

Reply:
Thanks, Tim.  I'll reword the message to make it more consistent, and, based on Peter's input below, more descriptive of the actual requirement.